Terminated/Killed integration-cli tests

On latest master code, as already outlined here https://github.com/docker/docker/pull/20509#issuecomment-186560841, integration-cli tests are Killed/Terminated. I cannot see anything weird in test.log nor any other log files. I'm experiencing this on my own laptop but PRs are also affected (see linked PR before). The tests usually fails at `TestDaemonLoggingDriverNone` but sometimes also on `Volumes*` tests.

```
PASS: docker_cli_daemon_test.go:575: DockerDaemonSuite.TestDaemonKeyGeneration	0.528s
PASS: docker_cli_daemon_test.go:594: DockerDaemonSuite.TestDaemonKeyMigration	0.524s
PASS: docker_cli_daemon_test.go:1045: DockerDaemonSuite.TestDaemonLinksIpTablesRulesWhenLinkAndUnlink	2.156s
PASS: docker_cli_daemon_test.go:498: DockerDaemonSuite.TestDaemonLogLevelDebug	0.518s
PASS: docker_cli_daemon_test.go:508: DockerDaemonSuite.TestDaemonLogLevelFatal	0.516s
PASS: docker_cli_daemon_test.go:431: DockerDaemonSuite.TestDaemonLogLevelWrong	5.501s
PASS: docker_cli_daemon_test.go:1154: DockerDaemonSuite.TestDaemonLoggingDriverDefault	1.066s
PASS: docker_cli_daemon_test.go:1196: DockerDaemonSuite.TestDaemonLoggingDriverDefaultOverride	1.063s
PASS: docker_cli_daemon_test.go:1217: DockerDaemonSuite.TestDaemonLoggingDriverNone	1.042s
/go/src/github.com/docker/docker/hack/make/test-integration-cli: line 18:  2829 Terminated              ( bundle .integration-daemon-start; bundle .integration-daemon-setup; bundle_test_integration_cli; bundle .integration-daemon-stop ) 2>&1
      2830                       | tee -a "$DEST/test.log"
```

```
---> Making bundle: .integration-daemon-setup (in bundles/1.11.0-dev/test-integration-cli)
---> Making bundle: .detect-daemon-osarch (in bundles/1.11.0-dev/test-integration-cli)
---> Making bundle: .ensure-emptyfs (in bundles/1.11.0-dev/test-integration-cli)
++++ tar -cC bundles/1.11.0-dev/test-integration-cli/emptyfs .
++++ docker load
---> Making bundle: .ensure-frozen-images (in bundles/1.11.0-dev/test-integration-cli)
++++ tar -cC /docker-frozen-images .
++++ docker load
++++ set +x
Untagged: hello-world:latest
---> Making bundle: .ensure-httpserver (in bundles/1.11.0-dev/test-integration-cli)
---> Making bundle: .ensure-syscall-test (in bundles/1.11.0-dev/test-integration-cli)
+ go test -check.f TestDaemonRestartWithSocketAsVolume -test.timeout=120m -check.v github.com/docker/docker/integration-cli
foobar [DOCKER_HOST=unix:///go/src/github.com/docker/docker/bundles/1.11.0-dev/test-integration-cli/docker.sock]
INFO: Testing against a local daemon
/go/src/github.com/docker/docker/hack/make/test-integration-cli: line 18:  3013 Terminated              ( bundle .integration-daemon-start; bundle .integration-daemon-setup; bundle_test_integration_cli; bundle .integration-daemon-stop ) 2>&1
      3014                       | tee -a "$DEST/test.log"
==================
WARNING: DATA RACE
Write by main goroutine:
  os.(*file).close()
      /usr/lib/golang/src/os/file_unix.go:129 +0x1be
  os.(*File).Close()
      /usr/lib/golang/src/os/file_unix.go:118 +0x88
  github.com/docker/docker/api/client.(*DockerCli).restoreTerminal()
      /home/runcom/koding/docker/.gopath/src/github.com/docker/docker/api/client/cli.go:104 +0xae
  github.com/docker/docker/api/client.(*DockerCli).CmdRun()
      /home/runcom/koding/docker/.gopath/src/github.com/docker/docker/api/client/run.go:287 +0x1832
  runtime.call64()
      /usr/lib/golang/src/runtime/asm_amd64.s:438 +0x3d
  reflect.methodValueCall()
      /usr/lib/golang/src/reflect/asm_amd64.s:29 +0x35
  main.main()
      /home/runcom/koding/docker/docker/docker.go:63 +0x680

Previous read by goroutine 22:
  os.(*File).read()
      /usr/lib/golang/src/os/file_unix.go:211 +0x57
  os.(*File).Read()
      /usr/lib/golang/src/os/file.go:95 +0xbc
  io.copyBuffer()
      /usr/lib/golang/src/io/io.go:381 +0x281
  io.Copy()
      /usr/lib/golang/src/io/io.go:351 +0x78
  github.com/docker/docker/api/client.(*DockerCli).holdHijackedConnection.func2()
      /home/runcom/koding/docker/.gopath/src/github.com/docker/docker/api/client/hijack.go:30 +0xf7

Goroutine 22 (running) created at:
  github.com/docker/docker/api/client.(*DockerCli).holdHijackedConnection()
      /home/runcom/koding/docker/.gopath/src/github.com/docker/docker/api/client/hijack.go:38 +0x14e
  github.com/docker/docker/api/client.(*DockerCli).CmdRun.func2()
      /home/runcom/koding/docker/.gopath/src/github.com/docker/docker/api/client/run.go:217 +0x15c
  github.com/docker/docker/pkg/promise.Go.func1()
      /home/runcom/koding/docker/.gopath/src/github.com/docker/docker/pkg/promise/promise.go:8 +0x2a
==================
Makefile:127: recipe for target 'test-integration-cli' failed
make: *** [test-integration-cli] Error 143
****
```
seems like the above fails frequently also

--- a/daemon/execdriver/native/driver.go
+++ b/daemon/execdriver/native/driver.go
@@ -157,6 +157,10 @@ func (d *Driver) Run(c *execdriver.Command, pipes *execdriver.Pipes, hooks execd
 	if err != nil {
 		return execdriver.ExitStatus{ExitCode: -1}, err
 	}
+
+	if err := cont.Start(p); err != nil {
+		return execdriver.ExitStatus{ExitCode: -1}, err
+	}
 	d.Lock()
 	d.activeContainers[c.ID] = cont
 	d.Unlock()
@@ -166,10 +170,6 @@ func (d *Driver) Run(c *execdriver.Command, pipes *execdriver.Pipes, hooks execd
 		}
 		d.cleanContainer(c.ID)
 	}()
-
-	if err := cont.Start(p); err != nil {
-		return execdriver.ExitStatus{ExitCode: -1}, err
-	}
 
 	//close the write end of any opened pipes now that they are dup'ed into the container
 	for _, writer := range writers {
@@ -301,6 +301,9 @@ func (d *Driver) Kill(c *execdriver.Command, sig int) error {
 	state, err := active.State()
 	if err != nil {
 		return err
+	}
+	if state.InitProcessPid == -1 {
+		return fmt.Errorf("avoid sending signal %d to container %s with pid -1", sig, c.ID)
 	}
 	return syscall.Kill(state.InitProcessPid, syscall.Signal(sig))
 }

